coda/coda_mdstore/tests/test_utils.py

```diff
@@ -1,14 +1,572 @@
+        assert isinstance(created_node, models.Node)
+
+    def test_created_node_attributes(self, rf):
+        node = factories.NodeFactory.build()
+        node_tree = presentation.nodeEntry(node)
+        node_xml = etree.tostring(node_tree)
+
+        request = rf.post('/', node_xml, 'application/xml')
+        created_node = presentation.createNode(request)
+
+        assert node.node_name == created_node.node_name
+        assert node.node_capacity == created_node.node_capacity
+        assert node.node_size == created_node.node_size
+        assert node.node_path == created_node.node_path
+        assert node.node_url == created_node.node_url
+
+        # Verify that the attribute exists, but do not attempt to guess
+        # the value.
+        assert hasattr(node, 'last_checked')
+
+
+class TestXmlToBagObject:
+
+    @pytest.fixture
+    def bag_xml(self):
+        xml = """
+            <bag:codaXML xmlns:bag="http://digital2.library.unt.edu/coda/bagxml/">
+            <bag:name>ark:/67531/coda2</bag:name>
+            <bag:fileCount>43</bag:fileCount>
+            <bag:payloadSize>46259062</bag:payloadSize>
+            <bag:bagitVersion>0.96</bag:bagitVersion>
+            <bag:lastStatus>fail</bag:lastStatus>
+            <bag:lastVerified>2015-01-01</bag:lastVerified>
+            <bag:bagInfo>
+                <bag:item>
+                    <bag:name>Bagging-Date</bag:name>
+                    <bag:body>2009-09-24</bag:body>
+                </bag:item>
+                <bag:item>
+                    <bag:name>Payload-Oxum</bag:name>
+                    <bag:body>46259062.43</bag:body>
+                </bag:item>
+            </bag:bagInfo>
+            </bag:codaXML>
+        """
+        return objectify.fromstring(xml)
+
+    def test_name_not_set(self, bag_xml):
+        del bag_xml.name
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag is None
+        assert bag_infos is None
+        assert error == "Unable to set 'name' attribute"
+
+    def test_name_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.name == bag_xml.name
+        assert error is None
+
+    def test_fileCount_not_set(self, bag_xml):
+        del bag_xml.fileCount
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag is None
+        assert bag_infos is None
+        assert error == "Unable to set 'files' attribute"
+
+    def test_fileCount_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.files == str(bag_xml.fileCount)
+        assert error is None
+
+    def test_payloadSize_not_set(self, bag_xml):
+        del bag_xml.payloadSize
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag is None
+        assert bag_infos is None
+        assert "Unable to set 'size' attribute" in error
```

Comment:
So unlike the section that report an error under a certain scenario, when the `payloadSize` is retrieved, if an error is caught, that error message is appended to the `error` return variable. 

Here is the [line](https://github.com/unt-libraries/coda/blob/master/coda/coda_mdstore/presentation.py#L219)

Since we don't care about the exceptions message, I chose to only test that the first part of the message is present.